it/controllers/agent/incomeSources/add/AddBusinessTradeControllerISpec.scala

```diff
@@ -0,0 +1,187 @@
+      enable(IncomeSources)
+      IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesResponse)
+
+      val formData: Map[String, Seq[String]] = {
+        Map(
+          BusinessTradeForm.businessTrade -> Seq("")
+        )
+      }
+
+      val result = IncomeTaxViewChangeFrontend.post("/income-sources/add/business-trade", clientDetailsWithConfirmation)(formData)
+
+      result should have(
+        httpStatus(BAD_REQUEST),
+        elementTextByID("business-trade-error")(messagesAPI("base.error-prefix") + " " +
+          messagesAPI("add-business-trade.form.error.empty"))
+      )
+    }
+  }
+
+  s"calling GET $changeBusinessTradeUrl" should {
+    "render the Change Business Trade page" when {
+      "User is authorised" in {
+        Given("I wiremock stub a successful Income Source Details response with multiple businesses and a uk property")
+        stubAuthorisedAgentUser(true)
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesResponse)
+
+        When(s"I call GET $changeBusinessTradeUrl")
+        val res = IncomeTaxViewChangeFrontend.getAddBusinessTrade(clientDetailsWithConfirmation)
+        verifyIncomeSourceDetailsCall(testMtditid)
+
+        res should have(
+          httpStatus(OK),
+          pageTitleAgent(pageTitleMsgKey),
+          elementTextByID("business-trade-hint")(pageHint),
+          elementTextByID("continue-button")(button)
+        )
+      }
+    }
+    "303 SEE_OTHER - redirect to home page" when {
+      "Income Sources FS disabled" in {
+        Given("I wiremock stub a successful Income Source Details response with no businesses or properties")
+        stubAuthorisedAgentUser(true)
+        disable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+        When(s"I call GET ${changeBusinessTradeUrl}")
+        val result = IncomeTaxViewChangeFrontend.getAddBusinessTrade(clientDetailsWithConfirmation)
+
+        result should have(
+          httpStatus(SEE_OTHER),
+          redirectURI(incomeSourcesUrl)
+        )
+      }
+    }
+  }
+
+  s"calling POST ${submitChangeBusinessTradeUrl}" should {
+    s"303 SEE_OTHER and redirect to $addBusinessAddressUrl" when {
+      "User is authorised and business trade is valid" in {
+        stubAuthorisedAgentUser(true)
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+        val formData: Map[String, Seq[String]] = {
+          Map(
+            BusinessTradeForm.businessTrade -> Seq("Test Business Trade")
+          )
+        }
+
+        When(s"I call POST ${changeBusinessTradeUrl}")
+        val result = IncomeTaxViewChangeFrontend.post("/income-sources/add/change-business-trade", clientDetailsWithConfirmation)(formData)
```

Comment:
Can you add a check for business trade name ? check with database using the session service